rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java

```diff
@@ -0,0 +1,700 @@
+                            }
+                            return tcpClient;
+                        }
+                );
+        builder.exchangeStrategies(ExchangeStrategies.builder()
+                .codecs(configurer -> configurer
+                        .defaultCodecs()
+                        .maxInMemorySize(config.getMaxInMemorySize()))
+                .build());
+
+        if (config.getHttpBasicAuthUsername() != null) {
+            builder.filter(ExchangeFilterFunctions
+                    .basicAuthentication(config.getHttpBasicAuthUsername(), config.getHttpBasicAuthPassword()))
+                    .build();
+        }
+
+        ReactorClientHttpConnector connector = new ReactorClientHttpConnector(httpClient);
+        webClient = builder.baseUrl(config.getBaseUrl()).clientConnector(connector).build();
+    }
+
+    @Override
+    public <T> ResponseEntity<T> get(String path, ParameterizedTypeReference<T> responseType) throws RestClientException {
+        return get(path, null, responseType);
+    }
+
+    @Override
+    public <T> ResponseEntity<T> get(String path, MultiValueMap<String, String> headers, ParameterizedTypeReference<T> responseType) throws RestClientException {
+        ClientResponse response;
+        try {
+            response = buildUri(webClient.get(), path)
+                    .headers(h -> {
+                        if (headers != null) {
+                            h.addAll(headers);
+                        }
+                    })
+                    .exchange()
+                    .block();
+            validateResponse(response, responseType);
+            return response.toEntity(responseType).block();
+        } catch (RestClientException ex) {
+            // Rethrow validation errors
+            throw ex;
+        } catch (Exception ex) {
+            throw new RestClientException("HTTP GET request failed", ex);
+        }
+    }
+
+    @Override
+    public void getNonBlocking(String path, Consumer<ClientResponse> onSuccess, Consumer<Throwable> onError) throws RestClientException {
+        getNonBlocking(path, null, onSuccess, onError);
+    }
+
+    @Override
+    public void getNonBlocking(String path, MultiValueMap<String, String> headers, Consumer<ClientResponse> onSuccess, Consumer<Throwable> onError) throws RestClientException {
+        try {
+            buildUri(webClient.get(), path)
+                    .headers(h -> {
+                        if (headers != null) {
+                            h.addAll(headers);
+                        }
+                    })
+                    .accept(config.getAcceptType())
+                    .exchange()
+                    .subscribe(onSuccess, onError);
+        } catch (Exception ex) {
+            throw new RestClientException("HTTP GET request failed", ex);
+        }
+    }
+
+    @Override
+    public <T> ObjectResponse<T> getObject(String path, Class<T> responseType) throws RestClientException {
```

Comment:
Maybe this is correct... but I am a little confused by the method signature. We are getting `Class<T>` but somehow returning `ObjectResponse<T>` - shouldn't this be a bit more explicit from the type-cast perspective?

Reply:
I do not see a problem, such construction is commonly used, see e.g.:
https://github.com/spring-projects/spring-boot/blob/07251388a92a0a0d444403203b464735ebb49fb9/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/webservices/client/WebServiceTemplateBuilder.java#L439

Comments from `Class.java`:
Type parameters:
<T> – the type of the class modeled by this Class object. For example, the type of String.class is Class<String>. Use Class<?> if the class being modeled is unknown.

We could use `ParameterizedTypeReference<T>` like in the generic cases, but I was trying to avoid it for Object requests / responses, because it's initialization is a bit clumsy. But it is for sure needed for the generic case which can have many levels of nested types.